src/configure

```diff
@@ -184,6 +189,29 @@ do
   --cudatk-dir=*)
     CUDATKDIR=`read_dirname $1`;
     shift ;; #CUDA is used in src/cudamatrix and src/nnet{,bin} only
+  --host=*)
```

Comment:
I see why I was confused about about this variable. You are using the host variable as a general prefix for the compiler tools -- not necessarily to specify the host architecture, right? From what I can tell the host variable represents target architecture when cross-compiling.

I don't like adding this variable to kaldi.mk when we are not cross-compiling. The kaldi.mk file is intended to be human-readable, so we are trying to keep it as simple and straightforward as possible. One way we can enable this behavior is to override the default compiler set in src/makefiles/*.mk when cross-compiling. I think overriding compiler tools via environment variables during the configure call will address this need in a more general way. So the configure call would be something like

`CXX=... ./configure ...`

We were already planning to add some partial support for this in #1249. For the moment we are exposing only CXX but we can also expose AR, AS and RANLIB if need be.

@danpovey Maybe we should do these changes after we merge #1249 if the plan is to merge it soon. On the other hand, it might be better to make these changes prior to C++11 requirement so people who prefer to stay on the 5.0 branch can also cross compile.